compiler/il/OMRBlock.cpp

```diff
@@ -1484,18 +1490,26 @@ OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyE
                   convertedValueNode = TR::Node::create(convertOpCode, 1, value);
                   }
                TR::SymbolReference *storedValueSymRef = createSymRefForNode(comp, methodSymbol, convertedValueNode, self()->getExit());
-               self()->getExit()->insertBefore(TR::TreeTop::create(comp,
-                                                                     TR::Node::createWithSymRef(value,
-                                                                        comp->il.opCodeForDirectStore(convertedValDataType),
+               TR::Node *storeNode = TR::Node::createWithSymRef(value, comp->il.opCodeForDirectStore(convertedValDataType),
                                                                         1,
                                                                         convertedValueNode,
-                                                                        storedValueSymRef)));
+                                                                        storedValueSymRef);
+#ifdef J9_PROJECT_SPECIFIC
+               if (convertedValueNode->getType().isBCD())
+                  storeNode->setDecimalPrecision(convertedValueNode->getDecimalPrecision());
+#endif
```

Comment:
Could we override `TR::Node::createWithSymRef` in OpenJ9 and do this there before we return `storeNode`? This will avoid polluting OMR code with BCD stuff since it doesn't exist in OMR.

Reply:
Looking at it, precision information is stored in value node. By overriding in OpenJ9 meant adding an over-ride in OpenJ9 for createWithSymRef with additional precision argument? This will be helpful, as I see couple of places in OMR where we do similar stuff for BCD node. This changes will create dependency on changes in OpenJ9. As changes are not that complicated won't take lot of time, I am thinking to make changes as part of separate pair PR that covers all the places in OMR. What do you think?

Reply:
No, you don't need to add any extra paramters. You have the node readily available which you alread ypass to the `createWithSymRef`, so you can just do the check there. For example the implementation can look like this:

```
static Node*
J9::Node::createWithSymRef(TR::Node * originatingByteCodeNode, TR::ILOpCodes op, uint16_t numChildren, TR::Node* first, TR::SymbolReference * symRef)
   {
   auto node = OMR::NodeConnector::createWithSymRef(...);

   if (op.isStore() && first->getType().isBCD())
      node->setDecimalPrecision(first->getDecimalPrecision());

   return node;
   }
```

Something along those lines. This seems simple enough of a change to just do it now rather than polluting OMR with new `J9_PROJECT_SPECIFIC` ifdefs and trying to circle back later to make the change.

Reply:
Actually this won't work for the next case, which can be either a load or a conversion. Both of these need to be handled. I'm ok merging this as the fix to eliminate these BCD precision setting calls needs to be a much wider effort.

Reply:
We should clean this up, but given the bugs we should fix this for now and then do the refactor. Can we have an issue for it if there isn't one?

Reply:
Opened up the issue: https://github.com/eclipse/openj9/issues/10290
As I have a fresh knowledge about this part, I am going to work on creating the API